public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -153,17 +172,26 @@ export default class PromQlLanguageProvider extends LanguageProvider {
       return [];
     }
 
-    // TODO #33976: make those requests parallel
-    await this.fetchLabels();
     this.metrics = (await this.fetchLabelValues('__name__')) || [];
-    await this.loadMetricsMetadata();
     this.histogramMetrics = processHistogramMetrics(this.metrics).sort();
-    return [];
+    return Promise.all([this.loadMetricsMetadata(), this.fetchLabels()]);
```

Comment:
I cannot see any direct reason to return an array here. Maybe I am missing something. Do we really return something here? Can't the return type be just `Promise<void>`>?

Reply:
I'm not sure if I understand? the `start` function in the parent `LanguageProvider` class defines start as a function that returns a promise of array of promises

Reply:
Yes, you are right.

Reply:
This should get us a performance improvement on its own also because we won't be waiting for the labels response to make the metadata request. For high cardinality/slow prometheus instances this should make the initial render a teensy bit faster, and then every subsequent load these should be cached for a bit too (at least metadata).